.github/workflows/run-test.yml

```
@@ -4,28 +4,34 @@ on:
   # push:
   #   branches: main
   #   paths-ignore: "docs/**"
-  pull_request_target:
+  pull_request:
     branches: main
     paths-ignore: "docs/**"
   workflow_dispatch:
 
 jobs:
   add-tag:
     runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.ref }}
+      cancel-in-progress: true
     steps:
       - name: Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
           token: ${{ github.token }}
-      - name: retag latest commit for testing
+      - name: Display source branch name
+        run: echo "Source branch name = ${{ github.event.pull_request.head.ref }}"
+      - name: Retag source branch HEAD commit for testing
         run: |
           git config user.name 'github-actions'
           git config user.email '41898282+github-actions[bot]@users.noreply.github.com'
           git tag --delete latest || true
           git push --delete origin latest || true
```

Stop arbitrarily ignoring any errors here.
```suggestion
          git tag --delete latest
          git push --delete origin latest
```